src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java

```diff
@@ -0,0 +1,655 @@
+import com.squareup.javapoet.TypeVariableName;
+
+import org.web3j.abi.EventEncoder;
+import org.web3j.abi.EventValues;
+import org.web3j.abi.FunctionEncoder;
+import org.web3j.abi.TypeReference;
+import org.web3j.abi.datatypes.DynamicArray;
+import org.web3j.abi.datatypes.Event;
+import org.web3j.abi.datatypes.Function;
+import org.web3j.abi.datatypes.StaticArray;
+import org.web3j.abi.datatypes.Type;
+import org.web3j.abi.datatypes.generated.AbiTypes;
+import org.web3j.crypto.Credentials;
+import org.web3j.protocol.ObjectMapperFactory;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.methods.request.EthFilter;
+import org.web3j.protocol.core.methods.response.AbiDefinition;
+import org.web3j.protocol.core.methods.response.Log;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.tx.Contract;
+import org.web3j.tx.TransactionManager;
+import org.web3j.utils.Collection;
+import org.web3j.utils.Console;
+import org.web3j.utils.Files;
+import org.web3j.utils.Strings;
+import org.web3j.utils.Version;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Future;
+
+import javax.lang.model.element.Modifier;
+
+import rx.functions.Func1;
+
+import static org.web3j.utils.Collection.tail;
+
+/**
+ * Generate Java Classes based on generated Solidity BIN and ABI.
+ */
+public class SolidityFunctionWrapper {
+
+    private static final String BINARY = "BINARY";
+    private static final String WEB3J = "web3j";
+    private static final String CREDENTIALS = "credentials";
+    private static final String TRANSACTION_MANAGER = "transactionManager";
+    private static final String INITIAL_VALUE = "initialEtherValue";
+    private static final String CONTRACT_ADDRESS = "contractAddress";
+    private static final String GAS_PRICE = "gasPrice";
+    private static final String GAS_LIMIT = "gasLimit";
+
+    private static final String CODEGEN_WARNING = "<p>Auto generated code.<br>\n" +
+            "<strong>Do not modify!</strong><br>\n" +
+            "Please use {@link " + SolidityFunctionWrapperGenerator.class.getName() +
+            "} to update.\n";
+
+    public void generateJavaFiles(String contractName, String bin, String abi, String destinationDirLocation, String basePackageName) throws IOException, ClassNotFoundException {
+        String className = Strings.capitaliseFirstLetter(contractName);
+
+        FieldSpec fieldSpec = FieldSpec.builder(String.class, "BINARY")
+                .addModifiers(Modifier.PRIVATE, Modifier.FINAL, Modifier.STATIC)
+                .initializer("$S", bin)
+                .build();
+
+        String javadoc = CODEGEN_WARNING + getWeb3jVersion();
+
+        TypeSpec.Builder classBuilder = TypeSpec.classBuilder(className)
```

Comment:
I see that a lot of the code here is from generateSolidityWrappers() in the original class. Please can you ensure it's presented in a similar manner. You appear to have got rid of a number of methods & pasted the method bodies in this method which makes the intentions of the code less clear.